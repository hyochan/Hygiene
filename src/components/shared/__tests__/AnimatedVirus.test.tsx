import 'react-native';

import * as React from 'react';

import {
  RenderAPI,
  cleanup,
  render,
} from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import Shared from '../AnimatedVirus';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: React.ReactElement;
let testingLib: RenderAPI;

describe('Rendering', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Shared {...props} />);
    testingLib = render(component);
  });

  afterEach(cleanup);

  it('renders without crashing', () => {
    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('Interaction', () => {
  beforeEach(() => {
    testingLib = render(component);
  });

  afterEach(cleanup);

  it('should simulate onClick', () => {
    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    // const btn = testingLib.queryByTestId('btn');
    // act(() => {
    //   fireEvent.press(btn);
    // });
    // expect(cnt).toBe(3);
  });
});
