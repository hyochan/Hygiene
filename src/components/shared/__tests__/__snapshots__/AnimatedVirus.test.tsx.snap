// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interaction should simulate onClick 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <Image
      source={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 114,
          "width": 114,
        }
      }
    />
  </View>
</View>
`;

exports[`Rendering renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <Image
      source={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 114,
          "width": 114,
        }
      }
    />
  </View>
</View>
`;
